tablo/interfaces/arcgis/views.py

```diff
@@ -259,44 +266,61 @@ def handle_request(self, request, **kwargs):
         except (DatabaseError, ValueError):
             return HttpResponseBadRequest(json.dumps({'error': 'Invalid request'}))
 
+        # Process limited query before calculating totals and counts
+        exceeded_limit = len(query_response) > int(limit)
+        query_response = query_response[:-1] if exceeded_limit else query_response
+
         if return_count_only:
-            response = query_response[0]
+            data = query_response[0]
+        elif return_format == 'csv':
+            header = list(query_response[0].keys())
+            header.sort()
+
+            data = [[h.strip('"').join('""') for h in header]] if offset == 0 else []
+            for item in query_response:
+                data.append([str(item[field]).strip('"').join('""') for field in header])
         else:
-            record_count = len(query_response)
-            exceeded_limit = record_count > int(limit)
-            query_response = query_response[:-1] if exceeded_limit else query_response
-
-            response = {
+            data = {
                 'count': len(query_response),
                 'total': len(query_response),
                 'exceededTransferLimit': exceeded_limit
             }
             if return_ids_only:
                 object_id_field = self.feature_service_layer.object_id_field
-                response['objectIdFieldName'] = object_id_field
-                response['objectIds'] = [feature[object_id_field] for feature in query_response]
+                data['objectIdFieldName'] = object_id_field
+                data['objectIds'] = [feature[object_id_field] for feature in query_response]
             else:
                 features = convert_wkt_to_esri_feature(query_response, self.feature_service_layer)
-                response.update({
+                data.update({
                     'count': len(features),        # Root level (source) features
                     'total': len(query_response),  # Total number of records queried
                     'fields': self.feature_service_layer.fields,
                     'geometryType': 'esriGeometryPoint',
                     'features': features
                 })
 
-                if response['total'] > response['count']:
-                    response['relatedFields'] = {
+                if data['total'] > data['count']:
+                    data['relatedFields'] = {
                         r.related_title: r.fields for r in self.feature_service_layer.relations
                     }
 
-        content = json.dumps(response, default=json_date_serializer)
-        content_type = 'application/json'
-        if self.callback:
-            content = '{callback}({data})'.format(callback=self.callback, data=content)
-            content_type = 'text/javascript'
+        if return_format == 'csv':
+            content = '\n'.join(','.join(col if isinstance(col, str) else str(col) for col in row) for row in data)
```

Comment:
I had looked at that before I chose this implementation. It makes assumptions about files, and would require more code (import and a `with` block for writing) that doesn't really serve us.

As this code is now, all values are wrapped in double quotes so that issue doesn't arise. However, I did check to see if double quotes are escaped within CSV values, and found they were not. The code wasn't breaking, and the CSV was still readable in Excel, but because this could cause issues when double quotes and commas appear in combination within a single value I decided to escape double quotes within each value by doubling them, so:

`A value with double quotes (") and a comma (,) in it`
...is output as ...
`"A value with double quotes ("") and a comma (,) in it"`

This meets CSV standards and I tested to make sure it opens correctly in Excel.